### PR TITLE
Improve and expand the tutorials

### PR DESCRIPTION
**Pull request by RobbeSneyders**
I made some further improvements to both the notebook and the README. I would like us to update all examples to match this.

Some changes:
- Make sure we link back to fondant and include a smaller version from the logo
- Remove unused art
- Have a main README which points to both a README and notebook tutorial
- Add automated prerequisite checks
- Expand the tutorial to include some customization of the pipeline

### CODE REVIEW

File: src/pipeline.ipynb [hunk 1]
```diff
@@ -0,0 +1,708 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "try:\n",
+    "    import fondant\n",
+    "except ImportError:\n",
+    "    logging.warning(\"Please install Fondant from the `requirements.txt` at the root of this repository\")"
```
Comment (mrchtr) [inline src/pipeline.ipynb]: I would add a cell to install the requirements.txt from the notebook. To make it really bulletproofed.

File: src/README.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+The pipeline is implemented in [pipeline.py](pipeline.py). Please have a look at the file so you 
+understand what is happening.
+
+For more details on the pipeline creation, you can have a look at the 
+[pipeline.ipynb](pipeline.ipynb) notebook which describes the process step by step.
+
+## Running the pipeline
+
+This pipeline will generate prompts, retrieve matching images in the laion dataset, download then
```
Comment (PhilippeMoussalli) [inline src/README.md]:
```suggestion
This pipeline will generate prompts, retrieve urls of matching images in the laion dataset, download them 
```

File: src/README.md [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+This pipeline will generate prompts, retrieve matching images in the laion dataset, download then 
+and generate corresponding captions and segmentations. If you added the optional `write_to_hf_hub` 
+component, it will write the resulting dataset to the HF hub.
+
+Fondant provides multiple runners to run our pipeline:
+- A Docker runner for local execution
+- A Vertex AI runner for managed execution on Google Cloud
+- A Kubeflow Pipelines runner for execution anywhere
+
+Here we will use the local runner, which utilizes docker-compose under the hood.
```
Comment (PhilippeMoussalli) [inline src/README.md]:
```suggestion
Here we will use the local runner, which utilizes Docker compose under the hood.
```

File: src/pipeline.py [hunk 4]
```diff
@@ -1,34 +1,29 @@
...
 logger = logging.getLogger(__name__)
 
-# General configs
-HF_USER = None  # Insert your huggingface username here
-HF_TOKEN = None  # Insert your HuggingFace token here
 BASE_PATH = "./data_dir"
-N_ROWS_TO_LOAD = 10  # Set to None to load all rows
-
-# Create data directory if it doesn't exist and if it's a local path
-if fsspec.core.url_to_fs(BASE_PATH)[0].protocol == ("file", "local"):
```
Comment (PhilippeMoussalli) [inline src/pipeline.py:17]: we only want to create a path if the base path is local. 

Maybe we can create it with fsspec but then we assume that the user has access to the base path

```python
fs = fsspec.core.url_to_fs(BASE_PATH)
fs.mkdir(BASE_PATH)
```

File: src/pipeline.py [hunk 5]
```diff
@@ -1,34 +1,29 @@
...
     pipeline_description="Pipeline that collects data to train ControlNet",
     base_path=BASE_PATH,
 )
 
-# Define component ops
+# Create a component operation from a local directory (custom component)
 generate_prompts_op = ComponentOp(
     component_dir="components/generate_prompts",
-    arguments={"n_rows_to_load": N_ROWS_TO_LOAD},
+    arguments={"n_rows_to_load": 10},
```
Comment (PhilippeMoussalli) [inline src/pipeline.py]:
```suggestion
    arguments={"n_rows_to_load": 10}, # Set to 10 for small scale testing, set to None to load all rows
```

File: src/pipeline.py
Comment (PhilippeMoussalli) [inline src/pipeline.py:1]: Maybe we can also add the GPU check to this file similar to the notebook?

**Review by PhilippeMoussalli — COMMENTED** [review]
Nice improvements! Left a few suggestions

File: src/pipeline.py (hunk 4 above)
Comment (RobbeSneyders) [reply src/pipeline.py:17]: I would assume a local path for the notebook and keep the code simple.

File: src/pipeline.py
Comment (RobbeSneyders) [reply src/pipeline.py:1]: It's in the README instead. I wouldn't add it to the `pipeline.py` as I don't think the pipeline definition should depend on the machine it's being compiled on.

Comment (PhilippeMoussalli) [reply src/pipeline.py:1]: alright makes sense

**Review by PhilippeMoussalli — APPROVED** [review]
Thanks Robbe! I will update the datacomp repo based on this

File: src/pipeline.ipynb [hunk 6]
```diff
@@ -0,0 +1,708 @@
...
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "from fondant.compiler import DockerCompiler\n",
+    "from fondant.runner import DockerRunner\n",
+    "\n",
+    "from pathlib import Path\n",
+    "\n",
+    "DockerCompiler().compile(pipeline=pipeline, output_path=\"docker-compose.yml\")\n",
```
Comment (mrchtr) [inline src/pipeline.ipynb:458]: Could we still add this to make it work on a M1 machine: 
```suggestion
     "import os\n",
     "os.environ["DOCKER_DEFAULT_PLATFORM"]="linux/amd64"\n",
    "DockerCompiler().compile(pipeline=pipeline, output_path=\"docker-compose.yml\")\n",
```

**Review by mrchtr — APPROVED** [review]
Thanks @RobbeSneyders. Last small nitpick for M1 users.

**Pull request merged**